OpenRA.Mods.Common/Traits/GainsExperience.cs

```diff
@@ -19,7 +19,7 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("This actor's experience increases when it has killed a GivesExperience actor.")]
 	public class GainsExperienceInfo : ITraitInfo, Requires<ValuedInfo>, Requires<UpgradeManagerInfo>
 	{
-		[FieldLoader.Require]
+		[FieldLoader.Require, UpgradeGrantedReference, UpgradeUsedReference] // HACK: Treat upgrades as used
```

Comment:
> Another option would be to add an IgnoreUsedUpgrades trait that pretends to use upgrades to silence warning about them being unused.

Yes, please.  I've been asking for this from the start of this upgrade linting saga.